### PR TITLE
Improve secure communication standards

### PR DESCRIPTION
**Pull request by Creatone**

### CODE REVIEW

File: wca/security.py [hunk 1]
```diff
@@ -134,3 +142,131 @@ def get_client_certs(self):
...
         return self.client_cert_path
+
+
+SECURE_CIPHERS = ':'.join([
+    'ECDHE-ECDSA-AES128-GCM-SHA256',
+    'ECDHE-RSA-AES128-GCM-SHA256',
+    'ECDH-ECDSA-AES128-GCM-SHA256',
+    'DHE-RSA-AES128-GCM-SHA256',
+    'AES128-GCM-SHA256',
+    'ECDHE-ECDSA-AES128-CCM',
```
Comment (Creatone) [inline wca/security.py]: Not supported in Centos 7

File: wca/security.py [hunk 2]
```diff
@@ -134,3 +142,131 @@ def get_client_certs(self):
...
+    'ECDHE-ECDSA-AES128-SHA256',
+    'ECDHE-RSA-AES128-SHA256',
+    'ECDHE-ECDSA-AES256-CCM',
```
Comment (Creatone) [inline wca/security.py]: Not supported in Centos 7

File: wca/security.py [hunk 3]
```diff
@@ -134,3 +142,131 @@ def get_client_certs(self):
...
+    'AES128-CCM',
```
Comment (Creatone) [inline wca/security.py]: Not supported in Centos 7

File: wca/security.py [hunk 4]
```diff
@@ -134,3 +142,131 @@ def get_client_certs(self):
...
+    'AES128-SHA256',
+    'AES256-CCM',
```
Comment (Creatone) [inline wca/security.py]: Not supported in Centos 7

File: wca/security.py [hunk 5]
```diff
@@ -134,3 +142,131 @@ def get_client_certs(self):
...
+    'AES256-SHA256',
+    'DHE-RSA-AES128-CCM',
```
Comment (Creatone) [inline wca/security.py]: Not supported in Centos 7

File: wca/security.py [hunk 6]
```diff
@@ -134,3 +142,131 @@ def get_client_certs(self):
...
+    'DHE-RSA-AES128-SHA256',
+    'DHE-RSA-AES256-CCM',
```
Comment (Creatone) [inline wca/security.py]: Not supported in Centos 7

File: wca/security.py [hunk 7]
```diff
@@ -134,3 +142,132 @@ def get_client_certs(self):
...
+
+
+def create_tcp_connection(module, address, timeout=None,
+                          use_ssl=False, ca=None, certfile=None,
+                          keyfile=None, keyfile_password=None,
+                          verify_certs=True, options=None, ciphers=None):
+    end = None
+    if timeout is None:
+        # thanks to create_connection() developers for
+        # this ugliness...
```
[the remark this answers is not in the document]
Comment (Creatone) [reply wca/security.py]: https://github.com/Creatone/workload-collocation-agent/blob/creatone/ssl-crypto/wca/security.py#L179

File: wca/security.py [hunk 8]
```diff
@@ -134,3 +142,132 @@ def get_client_certs(self):
...
+        timeout = module.getdefaulttimeout()
+    if timeout is not None:
```
[the remark this answers is not in the document]
Comment (Creatone) [reply wca/security.py]: https://github.com/Creatone/workload-collocation-agent/blob/creatone/ssl-crypto/wca/security.py#L179

File: tests/ssl/zookeeper/README.md [hunk 9]
```diff
@@ -0,0 +1,147 @@
...
+```
+#### Generate CA certs.
+```
+openssl genrsa -out /tmp/zk/root.key 2048
+```
+
+```
+openssl req -x509 -new -nodes -key /tmp/zk/root.key -sha256 -days 1024 -out /tmp/zk/root.crt -subj="/CN=root"
+```
+#### Import CA to zk truststore.
```
[the remark this answers is not in the document]
Comment (Creatone) [reply tests/ssl/zookeeper/README.md]: It's the like kubernetes and k8s. In my opinion if someone have any knowledge about Zookeeper it won't be a problem. Even start script is called "zkServer.sh".

File: wca/security.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (mmucek95) [reply wca/security.py]: But why ugliness?

**Pull request merged**